parsely/src/main/java/com/parsely/parselyandroid/HeartbeatIntervalCalculator.kt

```diff
@@ -0,0 +1,27 @@
+package com.parsely.parselyandroid
+
+import java.util.Calendar
+import kotlin.time.Duration.Companion.hours
+import kotlin.time.Duration.Companion.milliseconds
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+internal open class HeartbeatIntervalCalculator(private val clock: Clock) {
```

Comment:
1. Praise (❤️): Great work on this, I love how this class is now using duration and the `x.seconds/minutes/hours` approach, much more readable. Also, kudos for fixing the bugs of unwanted rounding (f273ee142c647cfc6a6ee24b3d68bf98acc2734d)! 🥇  

> It also covers the calculator with unit tests and changes the calculator implementation to use less error-prone kotlin.time.Duration class to make arithmetical operations between different times.

3. Suggestion (💡): When converting a Java class to Kotlin (98238156e20ae7b2bff6249a8d18af09f04dbd94), consider using the `Extra commit for .java > .kt renames` option while committing this change. This will make it easier for anyone reviewing this change later-on as it will show an actual diffing of files instead of showing the Java file being deleted and a new Kotlin file being added.

![image](https://github.com/Parsely/parsely-android/assets/9729923/01b90010-f73a-4ad8-a4d1-a95f61b8bfa2)

FYI: For more context on check [this discussion](https://github.com/wordpress-mobile/WordPress-Android/issues/17260#issuecomment-1549630646) I had with an external contributor a while back. Let me know your thoughts. 🙏

Reply:
Thanks for sharing about `Extra commit` option, TIL!